docs/resource/README.md

```diff
@@ -105,7 +105,27 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
 
-**[2]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122 for more recommendations).
+**[2]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4  [RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. Services aiming for reproducible UUIDs may also  use Version 5.
+SDKs are required to follow the following algorithm when generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this will
+  always be used and honored over generated IDs.
+- When the `container.id` resource attribute is provided, it should be used as the input
+  for generating a UUID v5.
+- When the SDK can detect it's running on Kubernetes or derivatives, such as by checking the
+  availability of a directory named `/var/run/secrets/kubernetes.io`, it should use
+  the contents of the file `/var/run/secrets/kubernetes.io/serviceaccount/namespace`,
+  along with the hostname and separated with a dot ("namespace.hostname") as the input
+  for generating a UUID v5.
+- When the SDK is running in an environment where a `/etc/machine-id` (see MACHINE-ID(5))
+  is available, the machine-id should be used as the input for generating a UUID v5.
+- When the SDK is running on a Windows environment and there's a reasonable way to read
+  registry keys for the SDK, the registry key
+  `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Cryptography\MachineGuid` can be used in a
+  similar way to Linux' machine-id above.
```

Comment:
Why not reference an ordered list of other (already existing) resource attributes that can be used, instead of duplicating information? Like this:
- [`container.id`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/container.md#container)
- [`k8s.namespace.name`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/k8s.md#namespace)/[`k8s.pod.name`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/k8s.md#pod)/[`k8s.container.name`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/k8s.md#container)
- [`host.id`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/host.md#host)

And what's the benefit we get from "obfuscating" it by generating UUIDs? Wouldn't a readable K8s `namespace/pod/container` name combination be of much more use than a UUID created from it?

Reply:
I'll enhance the `container.id` rule to use your other values, thank you for that suggestion!

The idea of hashing that is to not unintentionally leak information about the environment. For instance, the man page for machine-id states:

> This ID uniquely identifies the host. It should be considered "confidential", and must not be exposed in untrusted environments, in particular on the network.

Given that we don't need those values in clear text, and users would obtain them that way if they so wish, I believe they should be used in a non-reversible algorithm.